smart-py/objkt_swap_tests.py

```diff
@@ -0,0 +1,852 @@
+    scenario.verify(swap.data.swaps.contains(2) == False)
+
+    # TODO multiple edition tests
+
+@sp.add_test("Test mint")
+def test():
+    # init test and create html output
+    scenario = sp.test_scenario()
+    scenario.h1("Mint Objkt Test")
+
+    # init test values
+    seller = sp.test_account("seller")
+    manager = sp.test_account("manager")
+    curator = sp.test_account("curator")
+    objkt = sp.test_account("objkt123")
+    hdao = sp.test_account("hdao")
+
+    creator = sp.test_account("creator")
+
+    # TODO how do i turn this metadata into the expected binary object?
+    metadata = sp.record(
+        name = "test",
+        description = "test",
+        tags = [
+            'test'
+        ],
+        symbol = 'OBJKT',
+        artifactUri = "ipfs://test",
+        displayUri = "ipfs://test",
+        thumbnailUri = "ipfs://test",
+        creators = [
+            creator.address
+        ],
+        formats = [
+            {
+                "uri":"ipfs://test",
+                "mimeType":"image/png"
+            }
+        ],
+        decimals = 0,
+        isBooleanAmount = False,
+        shouldPreferSymbol = False
+    )
+
+    swap = OBJKTSwap(
+        objkt.address,
+        hdao.address,
+        manager.address,
+        metadata,
+        curator.address
+    )
+
+    scenario += swap
+
+    scenario.verify(swap.data.objkt_id == 152)
+
+    # try add 0 copies and fail
+    # the address and the sender are both the creator
+    scenario += swap.mint_OBJKT(
+        address = creator.address,
+        amount = 0,
+        royalties = 200,
+        # how to turn json into this?
+        metadata = sp.bytes('0x697066733a2f2f516d61794e7a7258547a354237577747574868314459524c7869646646504676775a377a364b7443377268456468')
+    ).run(
+        sender = creator.address,
+        valid = False
+    )
+
+    # hasn't changed
+    scenario.verify(swap.data.objkt_id == 152)
+
+    # try add more than 10k
+    scenario += swap.mint_OBJKT(
+        address = creator.address,
+        amount = 1000000,
+        royalties = 200,
+        # how to turn json into this?
+        metadata = sp.bytes('0x697066733a2f2f516d61794e7a7258547a354237577747574868314459524c7869646646504676775a377a364b7443377268456468')
+    ).run(
+        sender = creator.address,
+        valid = False
+    )
+
+    # hasn't changed
+    scenario.verify(swap.data.objkt_id == 152)
+
+    # no royalties are tracked yet
+    scenario.verify(swap.data.royalties.contains(152) == False)
+
+    # add an 1/1 objkt
+    scenario += swap.mint_OBJKT(
+        address = creator.address,
+        amount = 1,
+        royalties = 200,
+        # how to turn json into this?
+        metadata = sp.bytes('0x697066733a2f2f516d61794e7a7258547a354237577747574868314459524c7869646646504676775a377a364b7443377268456468')
+    ).run(
+        sender = creator.address,
+        valid = True
+    )
+
+    # royalty settings has been recorded for the new objkt
```

Comment:
Also, verify amount == 1

Reply:
not sure what you mean here?

Reply:
You are currently verifying the other mint parameter values get set correctly. Don't see verification that the amount == 1.